src/recommend.rs

```diff
@@ -0,0 +1,74 @@
+use std::ops::Index;
+
+use reqwest::Url;
+use serde_derive::{Deserialize, Serialize};
+
+type ConnectivityResponse = Vec<Nodes>;
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Nodes {
+    pub public_key: String,
+    pub channels: i64,
+    pub capacity: i64,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Node {
+    #[serde(rename = "public_key")]
+    pub public_key: String,
+    pub alias: String,
+    pub sockets: String,
+    #[serde(rename = "active_channel_count")]
+    pub active_channel_count: i64,
+    pub capacity: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Recommendations {
+    pub routing_node: Node,
+    pub edge_node: Node,
+}
+
+pub async fn get_recommended_channels() -> Result<Recommendations, RecommendedError> {
+    let url = "https://mempool.space/api/v1/lightning/nodes/rankings/connectivity".to_string();
+    // let url = Url::parse(&base_url).map_err(InternalRecommendedError::Url)?;
```

Comment:
Better to use the Url parser so that url is validated

Reply:
is github broken, i cant even see this line

Reply:
fixed